tests/integration/_conftest/spying_gateway_facade_factory.py

```diff
@@ -0,0 +1,35 @@
+from pathlib import Path
+from typing import Optional
+
+from starknet_py.net.gateway_client import GatewayClient
+
+from protostar.compiler.compiled_contract_reader import CompiledContractReader
+from protostar.starknet_gateway import GatewayFacade, GatewayFacadeFactory
+
+from .gateway_client_tx_interceptor import GatewayClientTxInterceptor
+from .transaction_registry import TransactionRegistry
+
+
+class SpyingGatewayFacadeFactory(GatewayFacadeFactory):
```

Comment:
Just out of curiosity: where'd you come up with such a name? Why "spying"?

Reply:
Because it uses `GatewayClientTxInterceptor` instead of `GatewayClient`. (Maybe `SpyingGatewayClient` would be a better name).  That client puts every transaction into `TransactionRegistry` which can be later used by tests to analyze what is actually sent.

I renamed it from `TestFriendyGatewayFacadeFactory`, because `spy` seems a better word than `test friendly`.

> Spies are stubs that also record some information based on how they were called.
https://martinfowler.com/bliki/TestDouble.html